http/sys_metrics.go

```diff
@@ -19,14 +27,20 @@ func handleMetricsUnauthenticated(core *vault.Core) http.Handler {
 		// Define response
 		resp, err := core.MetricsHelper().ResponseForFormat(format)
 		if err != nil {
-			w.WriteHeader(http.StatusBadRequest)
-			w.Write([]byte(err.Error()))
+			respondError(w, http.StatusBadRequest, err)
```

Comment:
This will change the Content-type to application/json.  Is that appropriate to do given that we may be disregarding the Accept header?

Reply:
If an error is returned, we never change the content type. We only return the error. I think that should be fine.

Reply:
But respondError calls logical.RespondError which does

```
	w.Header().Set("Content-Type", "application/json")
```

Reply:
Yes, correct. It doesn't change the way things are right now. A logical error response also uses `application/json` as content type.

Reply:
That's fine for Vault clients.  But if the client is Prometheus, it's not going to be happy getting an `application/json` response when it didn't specify that among the content types it accepts.  You can see e.g. in PrometheusResponse() that we're careful to set Content-Type to `text/plain` when returning a 400 because Prometheus isn't enabled.

Reply:
Thanks, @ncabatoff. I've updated the PR.